android/app/src/main/AndroidManifest.xml

```diff
@@ -44,43 +49,42 @@
-        <service
-            android:name="com.dieam.reactnativepushnotification.modules.RNPushNotificationListenerService"
-            android:exported="false" >
-            <intent-filter>
-                <action android:name="com.google.android.c2dm.intent.RECEIVE" />
-            </intent-filter>
-        </service>
+      <service android:name="com.dieam.reactnativepushnotification.modules.RNPushNotificationRegistrationService"/>
+      <service
+        android:name="com.dieam.reactnativepushnotification.modules.RNPushNotificationListenerService"
+        android:exported="false" >
+        <intent-filter>
+          <action android:name="com.google.android.c2dm.intent.RECEIVE" />
+        </intent-filter>
+      </service>
 
       <!-- background-location services (1 of 2) -->
-        <service android:name="com.transistorsoft.locationmanager.BackgroundGeolocationService" />
-        <service android:name="com.transistorsoft.locationmanager.LocationService" />
-        <service android:name="com.transistorsoft.locationmanager.ActivityRecognitionService" />
-        <service android:name="com.transistorsoft.locationmanager.geofence.GeofenceService" />
-        <service android:name="com.transistorsoft.locationmanager.scheduler.ScheduleService" />
-        <service android:name="com.transistorsoft.locationmanager.scheduler.ScheduleAlarmService" />
-        <receiver android:enabled="true" android:exported="false" android:name="com.transistorsoft.rnbackgroundgeolocation.BootReceiver">
-            <intent-filter>
-                <action android:name="android.intent.action.BOOT_COMPLETED" />
-            </intent-filter>
-        </receiver>
-
-        <!-- Google Maps API key -->
-        <meta-data
-          android:name="com.google.android.geo.API_KEY"
-          android:value="AIzaSyD3tJy2Eb_gp1KSxGsVKN-BY1XgytmHQTY" />
+      <service android:name="com.transistorsoft.locationmanager.BackgroundGeolocationService" />
+      <service android:name="com.transistorsoft.locationmanager.LocationService" />
+      <service android:name="com.transistorsoft.locationmanager.ActivityRecognitionService" />
+      <service android:name="com.transistorsoft.locationmanager.geofence.GeofenceService" />
+      <service android:name="com.transistorsoft.locationmanager.scheduler.ScheduleService" />
+      <service android:name="com.transistorsoft.locationmanager.scheduler.ScheduleAlarmService" />
+      <receiver android:enabled="true" android:exported="false" android:name="com.transistorsoft.locationmanager.BootReceiver">
+          <intent-filter>
+              <action android:name="android.intent.action.BOOT_COMPLETED" />
+          </intent-filter>
+      </receiver>
 
+      <!-- Google Maps API key -->
+      <meta-data
+        android:name="com.google.android.geo.API_KEY"
+        android:value="AIzaSyD3tJy2Eb_gp1KSxGsVKN-BY1XgytmHQTY" />
```

Comment:
Lets reference this value in build.gradle under buildTypes debug and release using the resValue method.

`resValue "string", "GOOGLE_API_KEY", GOOGLE_API_KEY`

This value can then be added to `~/.gradle/gradle.properties` on your local file system and kept from version control.


Finally AndroidManifest.xml would be changed to:

```
<meta-data
  android:name="com.google.android.geo.API_KEY"
  android:value="@string/GOOGLE_API_KEY" />
```

Let me know if this works